src/connection/server_support.c

```diff
@@ -764,6 +772,46 @@ css_process_get_eof_request (SOCKET master_fd)
 #endif
 }
 
+int
+css_process_master_hostname ()
+{
+  int hostname_length, error;
+
+  delete_master_hostname ();
```

Comment:
Host names can have maximum 255 characters (or you may use HOST_NAME_MAX)
it may be simpler to have a static char[] and allocate/dealocate

Reply:
yes, it simplified the code, thanks